oap-server/server-fetcher-plugin/kafka-fetcher-plugin/src/main/java/org/apache/skywalking/oap/server/analyzer/agent/kafka/KafkaFetcherHandlerRegister.java

```diff
@@ -132,16 +157,21 @@ public void start() {
     @Override
     public void run() {
         while (true) {
-            ConsumerRecords<String, Bytes> consumerRecords = consumer.poll(Duration.ofMillis(500L));
-            if (!consumerRecords.isEmpty()) {
-                Iterator<ConsumerRecord<String, Bytes>> iterator = consumerRecords.iterator();
-                while (iterator.hasNext()) {
-                    ConsumerRecord<String, Bytes> record = iterator.next();
-                    handlerMap.get(record.topic()).handle(record);
+            try {
+                ConsumerRecords<String, Bytes> consumerRecords = consumer.poll(Duration.ofMillis(500L));
+                if (!consumerRecords.isEmpty()) {
+                    Iterator<ConsumerRecord<String, Bytes>> iterator = consumerRecords.iterator();
+                    while (iterator.hasNext()) {
+                        ConsumerRecord<String, Bytes> record = iterator.next();
+                        executor.submit(() -> handlerMap.get(record.topic()).handle(record));
+                    }
+                    if (!enableKafkaMessageAutoCommit) {
```

Comment:
Change the condition to no `enable.auto.commit`. :)

Reply:
done.